tests/headers/array_pointers.h

```diff
@@ -0,0 +1,4 @@
+
+int test_fn(float a, int arr[20]);
```

Comment:
This should also add a test with typedefs.

Reply:
The problem with `typedefs` is that they are treated as regular arrays for some reason, I'm not sure if it's the right behavior or not. this means that this is kind of already the default thing but without a pointer:



```
typedef int arr1[20];
typedef int* arr2[20];
typedef int (*arr3)[20];
```

```

pub type arr1 = [::std::os::raw::c_int; 20usize];
pub type arr2 = [*mut ::std::os::raw::c_int; 20usize];
pub type arr3 = *mut [::std::os::raw::c_int; 20usize];

```

Reply:
Those typedefs look correct to me. What I meant is that something like: `void foo(arr1 a)` generates sane code. It should generate an in pointer without your patch, which is the right thing, but it should generate `a: *mut arr1` with your patch, not just `a: arr1`.

Reply:
Added. tell me if you're happy with the tests.